components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -335,11 +335,14 @@ sealed class ContentAction : BrowserAction() {
         ContentAction()
 
     /**
-     * Removes the [DownloadState] of the [ContentState] with the given [sessionId].
+     * Closes the [DownloadState.response] of the [ContentState.download] with the given [sessionId].
      */
-    data class ConsumeDownloadAction(val sessionId: String, val downloadId: String) :
-        ContentAction()
+    data class CloseDownloadResponse(val sessionId: String) : ContentAction()
```

Comment:
Does this require a separate action? Briefly scrolling through the PR it seems like this could be done when "consuming" the download? That would still sound logical to me: Consume the download and if there's a response then close it. But I haven't looked at all the surrounding code here, so you may know better. :)

Reply:
Thanks for taking a look :)
You can find more context about the separated action here -> https://github.com/mozilla-mobile/android-components/pull/9790#discussion_r584774846

Reply:
Yes, that confused me too in the original review :). Consuming here (also) means removing it from the tab and passing the download to the service for actual downloading where we still need the response.

One option to address this is to keep consume as we have it now and introduce a new use case and action for cancel. `CancelDownloadAction` and `useCases.cancelDownload` which would "consume" and close the response. Maybe that would make this clearer?

Reply:
I like the idea 👍🏽